package/MDAnalysis/analysis/pcz.py

```diff
@@ -0,0 +1,481 @@
+
+        if f_real < (self._u.trajectory.frame - 1):
+            self._u.trajectory.rewind()
+        while (self._u.trajectory.frame ) < f_real:
+            try:
+                self._u.trajectory.next()
+            except EOFError:
+                print 'End of file error trying to access frame ', frame
+                raise EOFError
+
+        crds = self._sel.ts._pos
+        if reference_structure is not None:
+            crds = lsqfit(crds, reference_structure)
+        return crds
+
+class Pca:
+    def __init__(self, universe, atom_selection=None, start=None, stop=None,
+    step=None, reference_coordinates=None, fitting_tolerance = 0.01,
+    max_iterations = 10, captured_variance = None):
+        """
+        Initialises a new Pca object with the data from the trajectory 
+        associated with the supplied universe, subject to the (optional) 
+        atom and frame selections. The selected atoms are least-squares 
+        fitted using an iterative procedure to remove global rotation and 
+        translation before the covariance matrix is calculated. 
+        If reference_coordinates, in the form of a [N, 3] numpy array (where 
+        N is the number of selected atoms) is provided, this will be used to 
+        initialise the least-squares fitting process, otherwise the first 
+        snapshot in the trajectory will be used.
+
+        :Arguments:
+          *universe*
+            :class:`MDAnalysis.Universe`
+
+          *atom_selection* 
+            a valid selection string for 
+            :meth:`MDAnalysis.core.AtomGroup.AtomGroup.select_atoms`
+
+          *start*
+            int, the index (zero-based) of the first frame in the trajectory
+            to include in the PC analysis
+
+          *stop*
+            int, the index (zero-based) of the last frame in the trajectory 
+            to include in the PC analysis
+
+          *step*
+            int, the increment between frames to be included in the analysis
+
+          *fitting_tolerance*
+            float, the iterative least-squares fitting process will terminate
+            when the average structure has an RMSD less than this from the
+            structure generated in the previous cycle (or if *max_iterations*
+            is reached first).
+
+          *max_iterations*
+            int, the maximum number of least-squares fitting cycles
+
+          *captured_variance*
+            int or float, if < 1 then enough eigenvectors will be kept to
+            explain *captured_variance* fraction of the total variance, if
+            >=1 (and integral), this is the number of eigenvectors that will
+            be kept. Providing this argument permits the precalculation of 
+            projections, which can improve performance.
+        """
+
+        # Step 1: Create the "Fasu" object from the selected trajectory data.
+        self._f = Fasu(universe, atom_selection=atom_selection, start=start, stop=stop, step=step)
```

Comment:
I also noticed that it has problems when `atom_selection` is `None`. I had to set it to `all` for the PCA class to be successfully initialized.